components/brave_wallet/browser/zcash/zcash_create_transparent_transaction_task_unittest.cc

```diff
@@ -0,0 +1,407 @@
+#include "brave/components/brave_wallet/browser/zcash/zcash_action_context.h"
+#include "brave/components/brave_wallet/browser/zcash/zcash_rpc.h"
+#include "brave/components/brave_wallet/browser/zcash/zcash_test_utils.h"
+#include "brave/components/brave_wallet/browser/zcash/zcash_transaction.h"
+#include "brave/components/brave_wallet/common/common_utils.h"
+#include "brave/components/brave_wallet/common/features.h"
+#include "components/sync_preferences/testing_pref_service_syncable.h"
+#include "content/public/test/browser_task_environment.h"
+#include "testing/gmock/include/gmock/gmock.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+using testing::_;
+using testing::SaveArg;
+
+namespace brave_wallet {
+
+constexpr char kAddress1[] = "t1WU75sSfiPbK5ez33uuhEbd9ZD3XNCxMRj";
+
+class MockZCashRPC : public ZCashRpc {
+ public:
+  MockZCashRPC() : ZCashRpc(nullptr, nullptr) {}
+  ~MockZCashRPC() override = default;
+
+  MOCK_METHOD2(GetLatestBlock,
+               void(const std::string& chain_id,
+                    GetLatestBlockCallback callback));
+};
+
+class MockZCashWalletService : public ZCashWalletService {
+ public:
+  MockZCashWalletService(base::FilePath zcash_data_path,
+                         KeyringService& keyring_service,
+                         std::unique_ptr<ZCashRpc> zcash_rpc)
+      : ZCashWalletService(zcash_data_path,
+                           keyring_service,
+                           std::move(zcash_rpc)) {}
+  MOCK_METHOD1(CreateTransactionTaskDone,
+               void(ZCashCreateTransparentTransactionTask* task));
+
+  MOCK_METHOD3(GetUtxos,
+               void(const std::string& chain_id,
+                    const mojom::AccountIdPtr& account_id,
+                    GetUtxosCallback));
+
+  MOCK_METHOD3(DiscoverNextUnusedAddress,
+               void(const mojom::AccountIdPtr& account_id,
+                    bool change,
+                    DiscoverNextUnusedAddressCallback callback));
+};
+
+class ZCashCreateTransparentTransactionTaskTest : public testing::Test {
+ public:
+  void SetUp() override {
+    feature_list_.InitAndEnableFeatureWithParameters(
+        features::kBraveWalletZCashFeature,
+        {{"zcash_shielded_transactions_enabled", "false"}});
+    ASSERT_TRUE(temp_dir_.CreateUniqueTempDir());
+    base::FilePath db_path(
+        temp_dir_.GetPath().Append(FILE_PATH_LITERAL("orchard.db")));
+
+    brave_wallet::RegisterProfilePrefs(prefs_.registry());
+    brave_wallet::RegisterLocalStatePrefs(local_state_.registry());
+
+    keyring_service_ =
+        std::make_unique<KeyringService>(nullptr, &prefs_, &local_state_);
+    keyring_service_->Reset();
+    keyring_service_->RestoreWallet(
+        "gallery equal segment repair outdoor bronze limb dawn daring main "
+        "burst "
+        "design palm demise develop exit cycle harbor motor runway turtle "
+        "quote "
+        "blast tail",
+        kTestWalletPassword, false, base::DoNothing());
```

Comment:
Can we have some of predefined test phrses used. Like constant for "abandon abandon... "